pkg/boot/kexec/elf_rel_load.go

```diff
@@ -0,0 +1,39 @@
+package kexec
+
+import (
+	"bytes"
+	"debug/elf"
+	"fmt"
+)
+
+// RelocateAndLoad does ...
+func RelocateAndLoad(purgatory []byte, purgatorySize uint, min uint, max uint, end int, flags uint32) (entry uintptr, err error) {
+	pgtory, err := elf.NewFile(bytes.NewReader(purgatory))
+	if err != nil {
+		return uintptr(0), fmt.Errorf("parse elf file from purgatory buffer: %v", err)
+	}
+
+	if pgtory.Version != elf.EV_CURRENT { // Should always be 1.
+		return uintptr(0), fmt.Errorf("unknown ELF version: %v", pgtory.Version)
+	}
+	if pgtory.Class != elf.ELFCLASS32 || pgtory.Class != elf.ELFCLASS64 {
+		return uintptr(0), fmt.Errorf("not a supported ELF class: %v", pgtory.Class)
+	}
+	if pgtory.Data != elf.ELFDATA2LSB || pgtory.Data != elf.ELFDATA2MSB {
+		return uintptr(0), fmt.Errorf("not a supported data type: %v", pgtory.Data)
+	}
+	if len(pgtory.Sections) == 0 {
+		return uintptr(0), fmt.Errorf("empty sections")
+	}
+
+	// TODO(10000TB): Un-comment and add relocation logic.
```

Comment:
I'm a bit confused by all this anyway. The purgatory should be written in position independent code. What is the relocation issue?